packages/salesforce-adapter/src/filters/territory.ts

```diff
@@ -32,13 +43,46 @@ const removeCustomFieldsFromTypes = (elements: Element[], typeNames: string[]):
     })
 }
 
+const isTerritoryRelatedChange = (change: Change): change is Change<InstanceElement> => (
+  isInstanceChange(change) && isAdditionOrModificationChange(change)
+   && territory2Types.some(typeName => isInstanceOfTypeChange(typeName)(change))
+)
+
+const setTerritoryDeployPkgStructure = (elements: InstanceElement[]): void => {
+  elements.forEach(element => {
+    const nestedDirsNames = [parentApiName(element)]
+    const { suffix } = element.type.annotations
+    let name = apiName(element)
+    if (territory2TypesToNestedDirName[metadataType(element)] !== undefined) {
+      const nestedDirName = territory2TypesToNestedDirName[metadataType(element)]
+      nestedDirsNames.push(nestedDirName)
+      name = apiName(element, true)
+    }
+    const contentFileName = `${name}${suffix === undefined ? '' : `.${suffix}`}`
+    element.annotate({ [CONTENT_FILENAME_OVERRIDE]: [...nestedDirsNames, contentFileName] })
```

Comment:
style nit to avoid changing things after they are defined.


```typescript
const territory2TypesToPath: Record<string, string[]> = {
  [TERRITORY2_RULE_TYPE]: ['rules'],
  [TERRITORY2_TYPE]: ['territories'],
}

```
and then
```suggestion
    const { suffix } = element.type.annotations
    const name = apiName(element, true)
    const contentPath = [
      parentApiName(element),
      ...territory2TypesToPath[metadataType(element)] ?? [],
      `${name}${suffix === undefined ? '' : `.${suffix}`}`
    ]
    element.annotate({ [CONTENT_FILENAME_OVERRIDE]: contentPath })
```

(this assumes we can always use `name = apiName(element, true)`, if we can't then it isn't clear this would be better)

Reply:
I like the workaround to the `if`, thanks!